modules/RTC/TraceablePeerConnection.js

```diff
@@ -2588,13 +2550,16 @@ TraceablePeerConnection.prototype.setSenderVideoConstraints = function(frameHeig
     if (frameHeight < 0) {
         throw new Error(`Invalid frameHeight: ${frameHeight}`);
     }
-
     const sourceName = localVideoTrack.getSourceName();
 
-    if (this._senderMaxHeights.get(sourceName) === frameHeight) {
+    this._senderMaxHeights.set(sourceName, frameHeight);
+
+    // Ignore sender constraints for the following cases -
+    // 1. If the media on the peerconnection is suspended (jvb conn when p2p is currently active).
+    // 2. If the client is already sending video of the requested resolution.
+    if (!this.videoTransferActive || this.tpcUtils.getEncodedResolution(localVideoTrack) === frameHeight) {
```

Comment:
Does this generate any problem if we set our consraint to a certain value but end up encoding different value, for whatever reason?

Reply:
I don't see how that can happen. The encode resolution depends on how many simulcast streams are enabled and this the only method that configures the encodings.

Reply:
Also, when the actual encode resolution changes because of cpu or bandwidth constraints, the encodings `scaleResolutionDownBy` value doesn't change. So this method only checks if the correct encodings are enabled based on the `frameHeight` received in the sender constraints.

Reply:
I can rename the method to getConfiguredEncodeResolution if that makes it more descriptive.

Reply:
👍